src/lib.rs

```diff
@@ -239,8 +239,21 @@ pub struct Engine {
 impl Engine {
     /// Create a new adblocking engine
     #[new]
-    pub fn from_rules(network_filters: Vec<String>) -> Self {
-        let engine = RustEngine::from_rules(&network_filters);
+    #[args(network_filters="None", load_network=true, load_cosmetic=false, debug=false)]
+    pub fn new(
+        network_filters: Option<Vec<String>>,
+        load_network: bool,
+        load_cosmetic: bool,
+        debug: bool,
+    ) -> Self {
+        let filters = network_filters.unwrap_or(Vec::new());
+        let engine = RustEngine::from_rules_parametrised(
+            &filters,
+            load_network,
+            load_cosmetic,
+            debug,
```

Comment:
Totally unrelated to the PR - but @ArniDagur, do you have any idea how the check warnings from GitHub Actions made it to the PR review? Surely there must be *something* responsible from parsing clippy's (?) output, but I have no idea what that is!

Reply:
I'm not sure which of the actions I'm using was responsible for that particular annotation, but you can look at the [source code for the `clippy-check` action](https://github.com/actions-rs/clippy-check/blob/master/src/check.ts#L342) to see an example of how it's done. Basically, `clippy` outputs JSON which the GitHub action processes and then uses to make an annotation using a GitHub API.